lib/jekyll/document.rb

```diff
@@ -306,21 +306,21 @@ def read(opts = {})
     end
 
     def post_read
-      if DATE_FILENAME_MATCHER =~ relative_path
-        m, cats, date, slug, ext = *relative_path.match(DATE_FILENAME_MATCHER)
-        merge_data!({
-          "slug" => slug,
-          "ext"  => ext
-        })
-        merge_data!({"date" => date}) if data['date'].nil? || data['date'].to_i == site.time.to_i
-        data['title'] ||= slug.split('-').select {|w| w.capitalize! || w }.join(' ')
-      end
-      populate_categories
-      populate_tags
+      if relative_path =~ DATE_FILENAME_MATCHER
+        cats, date, slug, ext = $1, $2, $3, $4
+        merge_data!("date" => date) if !data['date'] || data['date'].to_i == site.time.to_i
+        merge_data!("slug" => slug, "ext"  => ext)
 
-      if generate_excerpt?
-        data['excerpt'] ||= Jekyll::Excerpt.new(self)
+      elsif DATELESS_FILENAME_MATCHER =~ relative_path
+        cats, slug, ext = $1, $2, $3
```

Comment:
@parkr I don't, nice catch, I meant to actually move that down like I did with the `titleize`.